solana/solana-ibc/programs/solana-ibc/src/execution_context.rs

```diff
@@ -89,15 +93,17 @@ impl IbcStorage<'_, '_> {
             (head.timestamp_ns, head.block_height)
         };
 
+        let encoded_state = state.clone().encode_vec();
```

Comment:
How about changing `ConsensusState::encode_vec` to take `&self` argument so that we don’t need to `clone` here?

Reply:
`encode_vec` is defined in `ConsensusState` trait. But we can just add a method for `AnyConsensusState` which encodes by taking a ref.